api/src/main/java/org/openmrs/api/db/DelegatingFullTextSession.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db;
+
+import java.io.Serializable;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.lucene.search.Query;
+import org.hibernate.engine.spi.SessionDelegatorBaseImpl;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.search.FullTextQuery;
+import org.hibernate.search.FullTextSession;
+import org.hibernate.search.FullTextSharedSessionBuilder;
+import org.hibernate.search.MassIndexer;
+import org.hibernate.search.SearchFactory;
+import org.openmrs.api.APIException;
+import org.springframework.context.ApplicationEventPublisher;
+
+/**
+ * Custom implementation of the {@link FullTextSession} interface that acts a wrapper around a
+ * target FullTextSession instance, it actually delegates all the method calls directly to the
+ * target except for the {@link FullTextSession#createFullTextQuery(Query, Class[])} method where it
+ * first notifies registered listeners of the creation event before returning the newly created
+ * {@link FullTextQuery} object. The newly created query object and entity type are passed to the
+ * listeners wrapped in a {@link FullTextQueryAndEntityClass} object. <br>
+ * <br>
+ * An example use case is that a listener can enable/disable filters on the newly created query
+ * object.
+ */
+public class DelegatingFullTextSession extends SessionDelegatorBaseImpl implements FullTextSession {
+	
+	private static final Log log = LogFactory.getLog(DelegatingFullTextSession.class);
+	
+	private FullTextSession delegate;
+	
+	private ApplicationEventPublisher eventPublisher;
+	
+	public DelegatingFullTextSession(FullTextSession delegate, ApplicationEventPublisher eventPublisher) {
+		super((SessionImplementor) delegate, delegate);
+		this.delegate = delegate;
+		this.eventPublisher = eventPublisher;
+	}
+	
+	/**
+	 * @see FullTextSession#createFullTextQuery(Query, Class[])
+	 */
+	@Override
+	public FullTextQuery createFullTextQuery(Query luceneQuery, Class<?>... entities) {
+		if (entities.length > 1) {
+			throw new APIException("Can't create FullTextQuery for multiple persistent classes");
+		}
+		
+		if (log.isDebugEnabled()) {
```

Comment:
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-HowTo(Not)Log

Reply:
Did you take a look at the link that i gave?